build_tools/fetch_artifacts.py

```diff
@@ -6,58 +6,85 @@
 # but those artifacts may not have all required dependencies.
 
 import argparse
+import concurrent.futures
 import platform
-import shutil
-import subprocess
+import re
 import sys
+from urllib.request import urlopen, Request, urlretrieve, HTTPError
 
 GENERIC_VARIANT = "generic"
 PLATFORM = platform.system().lower()
+BUCKET_URL = "https://therock-artifacts.s3.us-east-2.amazonaws.com"
 
-
+# TODO(geomin12): switch out logging library
 def log(*args, **kwargs):
     print(*args, **kwargs)
     sys.stdout.flush()
 
 
-def check_aws_cli_available():
-    if not shutil.which("aws"):
-        log("[ERROR] AWS CLI not found in PATH.")
-        sys.exit(1)
-
-
-def s3_bucket_exists(run_id):
-    check_aws_cli_available()
-    cmd = [
-        "aws",
-        "s3",
-        "ls",
-        f"s3://therock-artifacts/{run_id}-{PLATFORM}",
-        "--no-sign-request",
-    ]
-    process = subprocess.run(cmd, check=False, stdout=subprocess.DEVNULL)
-    return process.returncode == 0
-
-
-def s3_exec(variant, package, run_id, build_dir):
-    check_aws_cli_available()
-    cmd = [
-        "aws",
-        "s3",
-        "cp",
-        f"s3://therock-artifacts/{run_id}-{PLATFORM}/{package}_{variant}.tar.xz",
-        str(build_dir),
-        "--no-sign-request",
-    ]
-    log(f"++ Exec [{cmd}]")
+def retrieve_s3_artifacts(run_id, amdgpu_family):
+    """Checks that the AWS S3 bucket exists and returns artifact names."""
+    index_page_url = f"{BUCKET_URL}/{run_id}-{PLATFORM}/index-{amdgpu_family}.html"
+    request = Request(index_page_url)
     try:
-        subprocess.run(cmd, check=True)
-    except Exception as ex:
-        log(f"Exception when executing [{cmd}]")
-        log(str(ex))
-
-
-def retrieve_base_artifacts(args, run_id, build_dir):
+        with urlopen(request) as response:
+            # from the S3 index page, we search for artifacts inside the a tags "<a href={TAR_XZ_NAME}>"
+            pattern = r'<a\s+[^>]*href=["\']([^"\']+)["\']'
+            artifact_files = re.findall(pattern, str(response.read()))
+            data = set()
+            for artifact in artifact_files:
+                # We only want to get .tar.xz files, not .tar.xz.sha256sum
+                if "sha256sum" not in artifact and "tar.xz" in artifact:
+                    data.add(artifact)
+            return data
+    except HTTPError as err:
+        if err.code == 404:
+            return None
```

Comment:
Should this log anything for 404? At least add a comment on why this returns `None` instead of some other logging or exception handling